kyuubi-common/src/main/scala/org/apache/kyuubi/Utils.scala

```diff
@@ -329,4 +329,6 @@ object Utils extends Logging {
    */
   def getContextOrKyuubiClassLoader: ClassLoader =
     Option(Thread.currentThread().getContextClassLoader).getOrElse(getKyuubiClassLoader)
+
+  def isOnK8s: Boolean = Files.exists(Paths.get("/var/run/secrets/kubernetes.io"))
```

Comment:
the solution is found at https://stackoverflow.com/a/49045575, and got confirmed by our internal K8s team. but I don't if there is an official recommended way to do that.